File: LeetCode/2.两数相加/Solution.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {
    fun addTwoNumbers(l1: ListNode?, l2: ListNode?): ListNode? {
        var node:ListNode? = ListNode(0)
        var n1:ListNode? = l1;
        var n2:ListNode? = l2;
        var p:ListNode? = node
        var cf = 0
        while (n1 != null || n2 != null || cf == 1){
            var val1 = if (n1 == null) 0 else n1?.`val`
            var val2 = if (n2 == null) 0 else n2?.`val`
            var sum = val1 + val2 + cf
            node?.`val` = sum % 10
            cf = sum / 10
            n1 = n1?.next
            n2 = n2?.next
            if (n1 != null || n2 != null || cf == 1){
                node?.next = ListNode(0)
                node = node?.next
            }
        }
        return p
    }
}

File: LeetCode/2.两数相加/Solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        node = ListNode()
        p = node
        cf = 0
        while l1 is not None or l2 is not None or cf == 1:
            if (l1 is not None):
                val1 = l1.val
                l1 = l1.next
            else:
                val1 = 0
            if (l2 is not None):
                val2 = l2.val
                l2 = l2.next
            else:
                val2 = 0
            sum = val1 + val2 + cf
            node.val = sum % 10
            cf = sum // 10
            if l1 is not None or l2 is not None or cf == 1:
                node.next = ListNode()
                node = node.next
        return p